library/src/main/java/org/xmtp/android/library/Conversation.kt

```diff
@@ -127,6 +155,8 @@ sealed class Conversation {
             is V2 -> {
                 conversationV2.prepareMessage(content = content, options = options)
             }
+
+            is Group -> throw XMTPException("Groups do not support prepared messages") // We return a encoded content not a preparedmessage which requires a envelope
```

Comment:
I'm not a kotlin expert and am not familiar with sealed classes. Are these methods being exposed directly to implementors, and are they required to know which methods they can/can't call on groups?

If so, can we find a way to use strong typing (for example, separate types for group and 1:1) where only the methods they can call are exposed?

Reply:
Ya I definitely have thought about this and maybe it can be spike/refactor. Since groups aren't being created users shouldn't experience this for now. But agree it almost feels like we need to return a type of either Conversation or Group instead of making Groups conform to conversations but it would be a little extra work. My hope is when we get rid of V2 this won't be a problem because most all the same functions should exist for both groups and 1to1 in that world.